apps/ledger-live-mobile/src/screens/ReceiveFunds/01-SelectCrypto.tsx

```diff
@@ -0,0 +1,152 @@
+// @flow
+import React, { useCallback, useMemo } from "react";
+import { Trans, useTranslation } from "react-i18next";
+import { StyleSheet, FlatList } from "react-native";
+import type {
+  CryptoCurrency,
+  TokenCurrency,
+} from "@ledgerhq/live-common/lib/types";
+import {
+  isCurrencySupported,
+  listTokens,
+  useCurrenciesByMarketcap,
+  listSupportedCurrencies,
+} from "@ledgerhq/live-common/lib/currencies";
+
+import {Flex } from "@ledgerhq/native-ui";
+import { useSelector } from "react-redux";
+import { Account, TokenAccount } from "@ledgerhq/live-common/src/types";
+import { makeEmptyTokenAccount } from "@ledgerhq/live-common/src/account";
+import { useRoute } from "@react-navigation/native";
+import { ScreenName } from "../../const";
+import { track, TrackScreen } from "../../analytics";
+import FilteredSearchBar from "../../components/FilteredSearchBar";
+import CurrencyRow from "../../components/CurrencyRow";
+import LText from "../../components/LText";
+import { flattenAccountsSelector } from "../../reducers/accounts";
+import { usePreviousRouteName } from "../../helpers/routeHooks";
+
+const SEARCH_KEYS = ["name", "ticker"];
+
+type Props = {
+  devMode: boolean,
+  navigation: any,
+  route: { params: { filterCurrencyIds?: string[] } },
+};
+
+const keyExtractor = (currency: CryptoCurrency | TokenCurrency) => currency.id;
+
+const renderEmptyList = () => (
+  <Flex px={6}>
+    <LText textAlign="center">
+      <Trans i18nKey="common.noCryptoFound" />
+    </LText>
+  </Flex>
+);
+
+const listSupportedTokens = () =>
+  listTokens().filter(t => isCurrencySupported(t.parentCurrency));
+
+const findAccountByCurrency = (accounts: (TokenAccount | Account)[], currency: CryptoCurrency | TokenCurrency) => accounts.filter((acc: TokenAccount | Account) =>
+    (acc.type === "Account" ? acc.currency?.id : acc.token.id) === currency.id
+  )
+
+export default function AddAccountsSelectCrypto({ navigation, route }: Props) {
+  const {t } = useTranslation();
+  const routerRoute = useRoute()
+  const { filterCurrencyIds = [] } = route.params || {};
```

Comment:
`[]` is always changing ref, making the useMemo always recalculating.

would suggest to have a global variable empty array and then you can do

```
const filterCurrencyIds = route.params?.filterCurrencyIds || emptyArray;
```

alternatively, the falsy logic can happen inside the memo (like basically nothing is filtered in this case

Reply:
updated it with useMemo :+1: